src/sceneHandler/scene-handler.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  StartScene,
+  LevelSelectionScreen,
+  GameplayScene,
+  LoadingScene,
+  LevelEndScene,
+} from "@scenes";
+import { DataModal, GameScore } from "@data";
+import { Debugger } from "@common";
+import {
+  LOADING_TRANSITION,
+  SCENE_NAME_START,
+  SCENE_NAME_LEVEL_SELECT,
+  SCENE_NAME_GAME_PLAY,
+  SCENE_NAME_LEVEL_END,
+  PWAInstallStatus,
+} from "@constants";
+import gameStateService from '@gameStateService';
+
+export class SceneHandler {
+  private scenes: {
+    LOADING_TRANSITION?: LoadingScene;
+    SCENE_NAME_START?: StartScene;
+    SCENE_NAME_LEVEL_SELECT?: LevelSelectionScreen;
+    SCENE_NAME_GAME_PLAY?: GameplayScene;
+    SCENE_NAME_LEVEL_END?: LevelEndScene;
+  };
+  private activeScene: null | StartScene | LevelSelectionScreen | GameplayScene | LevelEndScene;
+  public canvas: HTMLCanvasElement; //Remove and use DAO.
+  public data: DataModal; //Remove and use DAO.
+  public width: number; //Remove and use DAO.
+  public height: number; //Remove and use DAO.
+  public canavsElement: HTMLCanvasElement; //Remove and use DAO.
```

Comment:
can we correct the typo with a consistent naming "canvasElement"

Reply:
Corrected the typo from "canavsElement" to "canvasElement".